query/src/common/evaluator/scalar.rs

```diff
@@ -88,8 +93,8 @@ impl ScalarEvaluator {
                     Self::build_eval_tree(&comp.right)?,
                 ];
                 let func = FunctionFactory::instance().get(comp.op.to_func_name(), &[
-                    &comp.left.data_type(),
-                    &comp.right.data_type(),
+                    &wrap_nullable(&comp.left.data_type()),
```

Comment:
Because the build side column in outer join is nullable, here I directly wrapped nullable for left and right's `data_type` to make https://github.com/datafuselabs/databend/blob/main/common/datavalues/src/columns/series.rs#L41 happy. 

Do you have some better ideas? @leiysky

Reply:
I found maybe it's better to keep using `ExpressionEvaluator` here, which will infer data type from `DataSchema`.